### PR TITLE
cherrypicker: create comment for failed patch requests

### LINKED ISSUE #160: Provide additional response when external-plugin (cherrypick) fails
When using cherrypick plugin if there are any failures those needs to be surfaced on the PR with appropriate response. In this case when the diff for a PR is too large there is no response provided. Need to manually check the logs:

`{"component":"cherrypicker","error":"failed to get patch: status code 406 not one of [200], body: {\"message\":\"Sorry, the diff exceeded the maximum number of files (300). Consider using 'List pull requests files' API or locally cloning the repository instead.\",\"errors\":[{\"resource\":\"PullRequest\",\"field\":\"diff\",\"code\":\"too_large\"}],\"documentation_url\":\"https://docs.github.com/rest/pulls/pulls#list-pull-requests-files\"}","event-GUID":"3567fe10-178d-11ef-8d06-fcff71c12be5","event-type":"issue_comment","file":"sigs.k8s.io/prow/cmd/external-plugins/cherrypicker/server.go:151"`

### PR DESCRIPTION
**Pull request by AlexNPavel**
Fixes #160.

### CODE REVIEW

**Review by matthyx — APPROVED** [review]

File: cmd/external-plugins/cherrypicker/server.go [hunk 1]
```diff
@@ -426,7 +426,8 @@ func (s *Server) handle(logger *logrus.Entry, requestor string, comment *github.
 	// Fetch the patch from GitHub
 	localPath, err := s.getPatch(org, repo, targetBranch, num)
 	if err != nil {
-		return fmt.Errorf("failed to get patch: %w", err)
+		logger.WithError(err).Errorf("Failed to get patch for %s/%s#%d", org, repo, num)
+		return s.createComment(logger, org, repo, num, comment, "Failed to get PR patch from GitHub. This PR will need to be manually cherrypicked.")
```
Comment (stevekuznetsov) [inline cmd/external-plugins/cherrypicker/server.go:430]: Please embed the error (even if in a `<details>`) to short-circuit the support calls here - there might be other reasons for the error and we shouldn't need to go to server logs to see them.

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: cmd/external-plugins/cherrypicker/server.go (hunk 1 above)
Comment (AlexNPavel) [reply cmd/external-plugins/cherrypicker/server.go]: Added a details section to the comment.

File: cmd/external-plugins/cherrypicker/server.go [hunk 2]
```diff
@@ -426,7 +426,8 @@ func (s *Server) handle(logger *logrus.Entry, requestor string, comment *github.
 	// Fetch the patch from GitHub
 	localPath, err := s.getPatch(org, repo, targetBranch, num)
 	if err != nil {
-		return fmt.Errorf("failed to get patch: %w", err)
+		logger.WithError(err).Errorf("Failed to get patch for %s/%s#%d", org, repo, num)
+		return s.createComment(logger, org, repo, num, comment, "Failed to get PR patch from GitHub. This PR will need to be manually cherrypicked.\n<details><summary>Error message</summary>%v</details>")
```
Comment (petr-muller) [inline cmd/external-plugins/cherrypicker/server.go:430]: Are we missing a `fmt.Sprintf` to fill in  the `err` for `%v` there?

Comment (stevekuznetsov) [reply cmd/external-plugins/cherrypicker/server.go:430]: and the actual error value!

Comment (AlexNPavel) [reply cmd/external-plugins/cherrypicker/server.go]: Added the `fmt.Sprintf` and error value

**Comment by droslean** [comment]
/lgtm
/hold

Holding in case @petr-muller and @stevekuznetsov has more comment, but I think the PR should be ready to merge.

**Review by petr-muller — APPROVED** [review]
LGTM, thanks!

/hold cancel

**Pull request merged**